iteration/iteration_test.go

```diff
@@ -38,289 +38,273 @@ func (s *TestIterationRepository) TearDownTest() {
+		}
+		// add iteration to the second space from our fixture
+		e := repo.Create(context.Background(), &iteration.Iteration{
+			Name:    "Other Spring #2",
+			SpaceID: fxt.Spaces[1].ID,
+		})
+		require.Nil(t, e)
 
-	fxt := tf.NewTestFixture(s.T(), s.DB,
-		tf.Iterations(1,
-			tf.UserActive(false),
-			func(fxt *tf.TestFixture, idx int) error {
+		its, err := repo.List(context.Background(), fxt.Spaces[0].ID)
+		assert.Nil(t, err)
+		assert.Len(t, its, 3)
+	})
+
+	t.Run("load children", func(t *testing.T) {
+		fxt := tf.NewTestFixture(s.T(), s.DB,
+			tf.Iterations(3, func(fxt *tf.TestFixture, idx int) error {
 				i := fxt.Iterations[idx]
-				i.Name = "Sprint #24"
-				i.StartAt = &start
-				i.EndAt = &end
+				switch idx {
+				case 0:
+					i.Name = "Top level iteration"
+				case 1:
+					i.Name = "Level 1 iteration"
+					i.MakeChildOf(*fxt.Iterations[idx-1])
+				case 2:
+					i.Name = "Level 2 iteration"
+					i.MakeChildOf(*fxt.Iterations[idx-1])
+				}
 				return nil
-			},
-		),
-	)
-
-	i := *fxt.Iterations[0]
-
-	desc := "Updated item"
-	i.Description = &desc
-	updatedName := "Sprint 25"
-	i.Name = updatedName
-	// update iteration with new values of Name and Desc
-	updatedIteration, err := repo.Save(context.Background(), i)
-	require.Nil(t, err)
-	assert.Equal(t, updatedIteration.Name, updatedName)
-	assert.Equal(t, *updatedIteration.Description, desc)
-
-	changedStart := start.Add(time.Hour)
-	i.StartAt = &changedStart
-	changedEnd := start.Add(time.Hour * 2)
-	i.EndAt = &changedEnd
-	// update iteration with new values of StartAt, EndAt
-	updatedIteration, err = repo.Save(context.Background(), i)
-	require.Nil(t, err)
-	assert.Equal(t, changedStart, *updatedIteration.StartAt)
-	assert.Equal(t, changedEnd, *updatedIteration.EndAt)
+			}),
+		)
+		i1 := *fxt.Iterations[0]
+		i2 := *fxt.Iterations[1]
+		i3 := *fxt.Iterations[2]
+
+		// fetch all children of top level iteraiton
+		childIterations1, err := repo.LoadChildren(context.Background(), i1.ID)
+		require.Nil(t, err)
+		require.Equal(t, 2, len(childIterations1))
+		expectedChildIDs1 := []uuid.UUID{i2.ID, i3.ID}
+		var actualChildIDs1 []uuid.UUID
+		for _, child := range childIterations1 {
+			actualChildIDs1 = append(actualChildIDs1, child.ID)
+		}
+		assert.Equal(t, expectedChildIDs1, actualChildIDs1)
+
+		// fetch all children of level 1 iteraiton
+		childIterations2, err := repo.LoadChildren(context.Background(), i2.ID)
+		require.Nil(t, err)
+		require.Equal(t, 1, len(childIterations2))
+		expectedChildIDs2 := []uuid.UUID{i3.ID}
+		var actualChildIDs2 []uuid.UUID
+		for _, child := range childIterations2 {
+			actualChildIDs2 = append(actualChildIDs2, child.ID)
+		}
+		assert.Equal(t, expectedChildIDs2, actualChildIDs2)
+
+		// fetch all children of level 2 iteraiton
+		childIterations3, err := repo.LoadChildren(context.Background(), i3.ID)
+		require.Nil(t, err)
+		require.Equal(t, 0, len(childIterations3))
+
+		// try to fetch children of non-exisitng parent
```

Comment:
minor typo in comment: `non-exisitng`

Reply:
Thanks, done in https://github.com/fabric8-services/fabric8-wit/pull/1719/commits/2949ad3902c40f46bca32e444c41e2c67eb4f458